lib/helpers/cli.js

```diff
@@ -170,7 +170,11 @@ export function expandFileGlobs(workingDir, filePatterns, ignorePattern, glob =
     let isLiteralPath = !isGlob(pattern) && isFile(path.resolve(workingDir, pattern));
 
     if (isLiteralPath) {
-      let isIgnored = micromatch.isMatch(pattern, ignorePattern);
+      // If `--no-ignore-pattern` is passed, the ignorePatter is `[false]`.
+      let isIgnored =
+        ignorePattern?.length !== 1 &&
+        ignorePattern?.[0] !== false &&
+        micromatch.isMatch(pattern, ignorePattern);
```

Comment:
If this is a bug, can we extract it to a separate PR, and add a test for it?

Reply:
Fair point, will do.

Reply:
Bugfix pr: https://github.com/ember-template-lint/ember-template-lint/pull/2725

Reply:
Sorry for the long wait!